### ISSUE TITLE
Track expense - Unable to return to invite page after returning to confirmation page

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#40490]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.63-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**Issue found when executing PR:** https://github.com/Expensify/App/pull/38709
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to FAB > Track expense.
3. Create a track scan expense.
4. From the whisper, select Share it with my accountant.
5. Select a workspace > Click Submit.
6. Click back button on the Invite member RHP.
7. Click Submit.
8. Click back button twice.

## Expected Result:
In Step 7, user should be able to submit the expense, or user should be at least allowed to proceed to invite member page.
In Step 8. the RHP will close after clicking on the back button twice.

## Actual Result:
In Step 7, user is unable to proceed to invite member page after clicking on the Submit button.
In Step 8. not here RHP shows up.

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/6f2da910-d591-4078-9790-482a29ba1d98

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb

**Comment by thienlnam** [comment]
Will take this over since it's related to track - feel free to re-assign if you are interested

**Comment by thienlnam** [comment]
Doesn't look like it needs to be a blocker - and we can probably make this external

**Comment by DylanDylann** [comment]
RCA: In the confirmation page, when clicking submit we send the ShareTrackedExpense API before navigating to the invite page. When going back to the confirmation page, the request is submitted so we do nothing

@thienlnam From the RCA, could you help to confirm the expected on the OP?

> In Step 7, user should be able to submit the expense, or user should be at least allowed to proceed to invite member page.
In Step 8. the RHP will close after clicking on the back button twice.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After going back from the invite page, the confirmation page is shown with a request button disbaled.

### What is the root cause of that problem?
When we submit the request, we navigate to the invite page when it's a share track expense action.
https://github.com/Expensify/App/blob/9b2c5187e2203d83bba50bfa66936691788cbd18/src/libs/actions/IOU.ts#L3072-L3074

Going back from the invite page will navigate back user to the confirmation page because we never close it and pressing the request/submit button again won't do anything since `formHasBeenSubmitted` is true.
https://github.com/Expensify/App/blob/9b2c5187e2203d83bba50bfa66936691788cbd18/src/pages/iou/request/step/IOURequestStepConfirmation.tsx#L286-L290

### What changes do you think we should make in order to solve the problem?

Close all RHP before navigate to the invite page.
```
Navigation.dismissModal();
Navigation.navigate(ROUTES.ROOM_INVITE.getRoute(activeReportID ?? '', CONST.IOU.SHARE.ROLE.ACCOUNTANT));
```

**Comment by jjcoffee** [comment]
I agree that it makes the most sense if the back button closes the RHP, since we've already pushed the expense to the workspace and we may not always want to invite a new member so that screen needs to be dismissable. (Side note: The invite RHP seems to always show, even when an accountant/user has already been invited/added to the workspace).

It looks like there's another issue though, as after submitting the expense in step 5, we are not navigated to the workspace, but stay on the self-DM with the expense removed (the route has the workspace report ID, however). We only navigate _after_ inviting a member.

**Comment by DylanDylann** [comment]
> It looks like there's another issue though, as after submitting the expense in step 5, we are not navigated to the workspace, but stay on the self-DM with the expense removed (the route has the workspace report ID, however). We only navigate after inviting a member.

I agree with @jjcoffee's idea. But this change is so simple, should we update my proposal to cover this point

**Comment by jjcoffee** [comment]
I'm a bit torn here as whilst @DylanDylann's [proposal](https://github.com/Expensify/App/issues/40490#issuecomment-2065572639) was first, it is pretty scant on detail. My instinct is to go with @bernhardoj's [proposal](https://github.com/Expensify/App/issues/40490#issuecomment-2066019871) since it contains the actual implementation details.

:ribbon::eyes::ribbon: C+ reviewed

**Comment by DylanDylann** [comment]
The implementation for this idea is so minor and straightforward. It is the reason why I don't detail the solution in my proposal. But I test locally and It works well. One more thing, @bernhardoj's proposal will cause the flicker when navigating to invite page

**Comment by DylanDylann** [comment]
[Updated proposal](https://github.com/Expensify/App/issues/40490#issuecomment-2065572639) with detail implementation

**Comment by quinthar** [comment]
Great, what is the next step?  Who are we waiting on, to do what, and what is the ETA?

**Comment by DylanDylann** [comment]
@quinthar the solution is so straightforward. I can raise PR now if assigned

**Comment by jjcoffee** [comment]
> Great, what is the next step? Who are we waiting on, to do what, and what is the ETA?

@quinthar We're just waiting for @thienlnam to assign @bernhardoj per [this comment](https://github.com/Expensify/App/issues/40490#issuecomment-2066464876).

**Comment by bernhardoj** [comment]
PR is ready

cc: @jjcoffee

**Comment by jjcoffee** [comment]
PR approved by me, @thienlnam's turn! :rocket:

**Comment by garrettmknight** [comment]
Paid out - @jjcoffee can you complete the checklist when you get a chance?

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/38709/
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/38709/files#r1590707995
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A - just a missed test case**
- Determine if we should create a regression test for this bug. **Yes**
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. Open your self DM chat
2. Track a new expense. Wait until the concierge actionable whisper shows up
3. Choose Share it with my accountant
4. Select any workspace and Submit
5. Verify you are navigated to the invite page
6. Press the back button
7. Verify that the invite page is closed

Do we agree 👍 or 👎

**Comment by thienlnam** [comment]
Pending payment and addition of new QA steps cc @garrettmknight

**Comment by garrettmknight** [comment]
All set
